### PR TITLE
Allow to select an emoji + color as avatar [web client]

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Requires #9443 

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image | image
image | image

### 🚧 Tasks

- [ ] Visual check
- [ ] Code review

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/components/ConversationSettings/ConversationAvatarEditor.vue [hunk 1]
```diff
@@ -210,9 +246,38 @@ export default {
...
+					emoji: this.emojiAvatar,
+					color: this.backgroundColor ? this.backgroundColor.slice(1) : null,
+				})
+				this.loading = false
+				this.emojiAvatar = ''
+				this.backgroundColor = ''
+			} catch (e) {
+				showError(t('spreed', 'Error saving profile picture'))
+				this.loading = false
+			}
```
Comment (ShGKme) [inline src/components/ConversationSettings/ConversationAvatarEditor.vue:277]: `setConversationEmojiAvatarAction` doesn't throw any exceptions. 

Also, there it is not a `profile picture`, but a `conversation picture`.

File: src/components/ConversationSettings/ConversationAvatarEditor.vue [hunk 2]
```diff
@@ -210,9 +246,38 @@ export default {
...
+			try {
+				await this.$store.dispatch('setConversationEmojiAvatarAction', {
+					token: this.conversation.token,
+					emoji: this.emojiAvatar,
+					color: this.backgroundColor ? this.backgroundColor.slice(1) : null,
+				})
+				this.loading = false
+				this.emojiAvatar = ''
+				this.backgroundColor = ''
+			} catch (e) {
```
Comment (nickvergessen) [inline src/components/ConversationSettings/ConversationAvatarEditor.vue:274]: actually the error contains a translated message the user should see.
But I see it's missing from the documentation. Will fix

File: src/store/conversationsStore.js [hunk 3]
```diff
@@ -744,7 +745,18 @@ const actions = {
...
 
+	async setConversationEmojiAvatarAction(context, { token, emoji, color }) {
+		try {
+			const response = await setConversationEmojiAvatar(token, emoji, color)
+			const conversation = response.data.ocs.data
+			context.commit('addConversation', conversation)
+			showSuccess(t('spreed', 'Conversation picture set'))
+		} catch (error) {
+			console.error(error)
+			showError(t('spreed', 'Could not set the conversation picture'))
```
Comment (nickvergessen) [inline src/store/conversationsStore.js:758]: Should show the error from the response:
https://github.com/nextcloud/spreed/pull/9443/files#diff-d2141d061bc161b347303058b3e5128b4f3dddbd4999da98444d2ef2574d119fR50

Comment (nickvergessen) [reply src/store/conversationsStore.js:758]: Actually this should not showError as the UI is already doing it?
And as per recent discussion you should not catch the error here, but the frontend code should deal with it.

Comment (Antreesy) [reply src/store/conversationsStore.js:758]: Picture | Emoji
-- | --
image | image

**Comment by Antreesy** [comment]
Rebased onto updated master for supporting by backend

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**